Sources/Shared/Models/VIMVideo.m

```diff
@@ -287,6 +294,16 @@ - (void)parseConnections
     self.connections = connections;
 }
 
+- (void)parsePublishJobConnection
+{
+    NSDictionary *dictionary = [[self.metadata valueForKey:@"connections"] valueForKey:@"publish_to_social"];
+    if (dictionary != nil) {
+        VIMObjectMapper *mapper = [[VIMObjectMapper alloc] init];
+        [mapper addMappingClass: [PublishJobConnection class] forKeypath:@""];
+        self.publishJobConnection = [mapper applyMappingToJSON:dictionary];
```

Comment:
I tried something like this originally, but ran into an issue when trying to cast the returned connection in Swift. `connection(withName:)` will still always return a `VIMConnection`, even if we try to cast it as `PublishJobConnection`, because the parent class doesn't know anything about its subclass.

But it sounds like what you're suggesting is that these new properties will just be nil for any other `VIMConnection`. Is that right? Would that mean adding those properties to `VIMConnection` directly, instead of creating the `PublishJobConnection` subclass?

Reply:
Ah ok, I see what you mean about the casting issue. Makes sense that you need to keep this separate. 

For the second part, I guess that is an approach we could take? But I think a unique connection object makes more sense at this point. 

If other consumers like Anton and Liron feel this is too divergent from the existing API then I would understand trying to keep the existing format but I'm not aware of any use cases where we need to iterate through all connections and do something with all of them. 

I'm down to keep as is barring cases mentioned above.